src/EFRepository.Generator/ExtensionMethodGenerator.cs

```diff
@@ -1,237 +1,388 @@
-		/// </summary>
-		/// <param name=""start"">The <see cref=""DateTime""/> that should be before {member.Name}</param>
-		/// <param name=""end"">The <see cref=""DateTime""/> that should be after {member.Name}</param>
-		public static IQueryable<{dbSetClass}> Where{member.Name}IsBetween(this IQueryable<{dbSetClass.Name}> queryable, DateTime start, DateTime end)
-		{{
-			if (queryable == null)
-				return queryable;
-
-			return queryable.Where(n => n.{member.Name} > start && n.{member.Name} < end);
-		}}");
-			}
-
-		}
-
-		builder.AppendLine("	}")
-		.AppendLine("}");
-
-		return (fileName, builder.ToString());
-
-		string CreateMethod(string type, string memberName)
-		{
-			return $@"		/// <summary>
-		/// Filter the <see cref=""IQueryable""/> of {dbSetClass.Name} by {memberName}
-		/// </summary>
-		/// <param name=""value"">The {type} that {memberName} should be equal to</param>
-		public static IQueryable<{dbSetClass.Name}> By{memberName}(this IQueryable<{dbSetClass.Name}> queryable, {type} value)
-		{{
-			if (queryable == null)
-				return queryable;
-
-			return queryable.Where(n => n.{memberName} == value);
-		}}";
-		}
-	}
-}
-
-public record SourceRecord
-{
-	public SemanticModel Model { get; init; }
-	public INamedTypeSymbol Symbol { get; init; }
-	public TypeDeclarationSyntax Syntax { get; init; }
-}
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
+
+namespace EFRepository.Generator;
+
+[Generator]
+public class ExtensionMethodGenerator : ISourceGenerator
+{
+
+	public void Initialize(GeneratorInitializationContext context)
+	{
+		context.RegisterForSyntaxNotifications(() => new SyntaxReceiver());
+	}
+
+	public void Execute(GeneratorExecutionContext context)
+	{
+		if (context.SyntaxReceiver is not SyntaxReceiver receiver)
+			return;
+
+		var options = ((CSharpCompilation)context.Compilation).SyntaxTrees[0].Options as CSharpParseOptions;
+		var compilation = context.Compilation;
+
+		var dbContextSymbol = compilation.GetTypeByMetadataName("Microsoft.EntityFrameworkCore.DbContext");
+
+		if (dbContextSymbol == null)
+			dbContextSymbol = compilation.GetTypeByMetadataName("Microsoft.EntityFramework.DbContext");
+
+		if (dbContextSymbol == null)
+			return;
+
+		var dbSetSymbol = compilation.GetTypeByMetadataName("Microsoft.EntityFrameworkCore.DbSet`1");
+
+		if (dbSetSymbol == null)
+			dbSetSymbol = compilation.GetTypeByMetadataName("Microsoft.EntityFramework.DbSet");
+
+		if (dbSetSymbol == null)
+			return;
+
+		var dbContexts = receiver.Classes.Select(c =>
+		{
+			var sourceModel = compilation.GetSemanticModel(c.SyntaxTree);
+			var sourceSymbol = sourceModel.GetDeclaredSymbol(c);
+
+			if (sourceModel == null || sourceSymbol == null)
+				return null;
+
+			return new SourceRecord
+			{
+				Model = sourceModel,
+				Symbol = sourceSymbol,
+				Syntax = c
+			};
+		})
+		.Where(c => c != null)
+		.Where(c =>

```

Comment:
This formatting doesn't follow our standards of having the first item on the same line like for XAML and HTML, but it's a minor thing.